packages/modules/e3dc/counter.py

```diff
@@ -37,13 +48,13 @@ def __init__(self,
         self.__store = get_counter_value_store(self.component_config.id)
         self.component_info = ComponentInfo.from_component_config(self.component_config)
 
-    def update(self, client: modbus.ModbusTcpClient_):
+    def update(self, client: modbus.ModbusTcpClient_) -> None:
         power, powers = read_counter(client)
         imported, exported = self.sim_counter.sim_count(power)
         counter_state = CounterState(
             imported=imported,
             exported=exported,
-            powers=powers,
+            powers=cast(List[float], powers),
```

Comment:
Das stellt mypy glücklich, macht aber nicht das "richtige". `powers` ist eine `List[int]` und du sagst hiermit "liebes mypy, ich versichere dir es sind floats". Sind es aber nicht.

Was du machen könntest ist:
```python
powers=list(map(float, powers))
```
Dann sind es tatsächlich floats.

Allerdings denke ich, dass `CounterState` auch ganz wunderbar mit `int` zurecht kommt. Insofern kann es auch die Lösung sein `CounterState` anzupassen, so dass es `Union[int, float]`, also beides akzeptiert. Ich fände es auch OK, wenn du den Fehler, den mypy hier wirft für den Moment ignorierst. Das gehört zu den vielen Sachen, die hier noch optimiert werden können bis mypy großflächig zum Einsatz kommen kann.

Reply:
gelöst in https://github.com/snaptec/openWB/pull/2510